woodwork/table_accessor.py

```diff
@@ -164,6 +164,37 @@ def set_index(self, new_index):
             _check_index(self._dataframe, self._schema.index)
         self._set_underlying_index()
 
+    def set_types(self, logical_types=None, semantic_tags=None, retain_index_tags=True):
+        """Update the logical tpye and semantic tags for any columns names in the provided types dictionaries,
+        updating the Woodwork typing information for the DataFrame.
+
+        Args:
+            logical_types (dict[str -> str], optional): A dictionary defining the new logical types for the
+                specified columns.
+            semantic_tags (dict[str -> str/list/set], optional): A dictionary defining the new semantic_tags for the
+                specified columns.
+            retain_index_tags (bool, optional): If True, will retain any index or time_index
+                semantic tags set on the column. If False, will replace all semantic tags any time a column's
+                semantic tags or logical type changes. Defaults to True.
+        """
+        logical_types = logical_types or {}
+        logical_types = {col_name: _get_column_logical_type(None, ltype, col_name) for col_name, ltype in logical_types.items()}
+
+        original_index = self._schema.index
+        self._schema.set_types(logical_types=logical_types,
+                               semantic_tags=semantic_tags,
+                               retain_index_tags=retain_index_tags)
+        # go through changed ltypes and update dtype if necessary
+        for col_name, logical_type in logical_types.items():
+            series = self._dataframe[col_name]
+            updated_series = _update_column_dtype(series, logical_type)
+            if updated_series is not series:
+                self._dataframe[col_name] = updated_series
+
+        # if index has been reset, reset underlying index of DataFrame
+        if original_index is not None and self._schema.index is None:
+            self._set_underlying_index()
```

Comment:
Felt it was important to follow our behavior of resetting the underlying index when it's removed, but this isn't getting done for `reset_semantic_tags` or `remove_semantic_tags` since they only exist on the Schema class, and this isn't actually behavior that we want (#597 ), so maybe it's better to not have this here at all.

Reply:
TBH the underlying index behavior is getting me a bit confused. I'd vote we have some good discussion about how we really want it to work in all these different scenarios and tackle it separate from this issue.

Reply:
I think that's very fair

Reply:
Going to remove this behavior so that there's not inconsistency with index removal via tags